discretize/base/base_mesh.py

```diff
@@ -169,7 +169,7 @@ def to_dict(self):
                 out[item] = attr
         return out
 
-    def __eq__(self, other):
+    def equals(self, other):
```

Comment:
Out of interest - what is the reasoning (and implication) of the rename? Would that mean that `meshA == meshB` could return something different than `meshA.equals(meshB)`?

Reply:
It had to deal with hashing. Essentially it’s this: if an object implements `__eq__` it must also implement `__hash__` to be used as a key (and this happens quite often in SimPEG). Because our meshes are Mutable objects (I.e. `origin` can be changed) the hash value could change over its lifetime and this lead to undefined behavior. Thus we deemed it best to not worry about implementing those two and allowing default python behavior of equality and hashing (which default to `x is y` and `id(x)` respectively)